test/e2e/workflow/deployment_test.go

```diff
@@ -277,6 +281,24 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			CreateConfig(gvk, configSpec)
 			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, 30*time.Second)
 		})
+		Context("and checking CNAO prometheus endpoint", func() {
+			var err error
+			var scrapedData string
+			BeforeEach(func() {
+				scrapedData, err = GetScrapedDataFromMonitoringEndpoint()
+				Expect(err).ToNot(HaveOccurred())
+			})
+			It("Should report the expected metrics", func() {
+				expectedmetricValueMap := map[string]string{
+					"kubevirt_cnao_ready": "1",
+				}
+
+				for metricName, expectedValue := range expectedmetricValueMap {
+					Expect(scrapedData).To(ContainSubstring(metricName), fmt.Sprintf("metric %s does not appear in endpoint scrape", metricName))
```

Comment:
isnt it redundant, as the next line will include `metricName` ?

Reply:
I agree, changed the test, tell me what you think

Reply:
I think that 2nd line was enough, no need to add the parsing
as the only line that match VAR_NAME VALUE is the one we need (beside *)

`Expect(scrapedData).To(ContainSubstring(fmt.Sprintf("%s %s", metricName, expectedValue)), fmt.Sprintf("metric %s does not have the correct value %s", metricName, expectedValue))`

maybe nice to have is something like
var re = regexp.MustCompile(`^VAR_NAME VALUE$`)
so its the exact line as a whole, this way it won't match a prefix of other variable (*)

wdyt?

Reply:
scrapedData is huge, I want to avoid `Expect`ing the entire scrapedData string, as the output on the test failure will be too tiresome to debug, and also might be trancated by gomega..